### PR TITLE
Features

### PR DESCRIPTION
**Pull request by Tresor11**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  validates :name, presence: { message: 'Please provide your full name' }
+  validates :password, confirmation: true, length: { minimum: 8, too_short: 'Password must be at least 8 characters long' }
+  validates :email, uniqueness: true
+  validates :user_name, uniqueness: { case_senstive: false }
```
Comment (efrapp) [inline app/models/user.rb]: What is the purpose of this `user_name` field? you already have a `name` field above. Also, you are managing the uniqueness of the user record with the `email`.

Comment (Tresor11) [reply app/models/user.rb]: we have made the required changes  :smiley:

**Comment by Tresor11** [comment]
we have made the required changes by correction our migration files and our model's validations :smiley:

File: app/models/user.rb [hunk 2]
```diff
@@ -5,6 +5,5 @@ class User < ApplicationRecord
   has_many :comments
   validates :name, presence: { message: 'Please provide your full name' }
   validates :password, confirmation: true, length: { minimum: 8, too_short: 'Password must be at least 8 characters long' }
-  validates :email, uniqueness: true
-  validates :user_name, uniqueness: { case_senstive: false }
+  validates :email, uniqueness: { message: 'This email already exists' }
```
Comment (samgaco) [inline app/models/user.rb]: :+1:

**Pull request merged**